### ISSUE TITLE
Cannot receive any new message after finish subscription to updates and subscribe again

### ISSUE THREAD

**Issue by jiayu-hotstar** [shamblett/mqtt_client#177]
An reproduce code might be:

```
var client = MqttServerClient('...')
final connMsg = MqttConnectMessage().withClientIdentiifer().authenticateAs(...)
client.connect()

let sub1 = client.updates(callback1);
sub1.cancel();

// callback2 will never receive new message
let sub2 = client.updates(callback2);

**Comment by jiayu-hotstar** [comment]
https://github.com/shamblett/mqtt_client/pull/176#issue-405621183

**Comment by jiayu-hotstar** [comment]
https://github.com/shamblett/mqtt_client/pull/176

**Comment by shamblett** [comment]
Merged the pull request, I'll re publish shortly

**Comment by shamblett** [comment]
Client updated and re published at 7.0.0.
